File: sdk-python/tests/iwfcompat/shared.py
# Copyright (c) 2026 Super Durable, Inc.
#
# Licensed under the Super Durable Source License 1.0.
# You may not use this file except in compliance with the License.
# See the LICENSE file in the repository root.
#
# SPDX-License-Identifier: LicenseRef-Super-Durable-1.0

from dataclasses import dataclass

from dex import Context, Step, StepDecision, graceful_complete


@dataclass
class ModelInput:
    value: int = 0


class CompleteStringStep(Step[str]):
    def execute(self, context: Context, input: str) -> StepDecision:
        del context
        return graceful_complete(input)

File: sdk-typescript/test/iwfcompat/shared.ts
// Copyright (c) 2026 Super Durable, Inc.
//
// Licensed under the Super Durable Source License 1.0.
// You may not use this file except in compliance with the License.
// See the LICENSE file in the repository root.
//
// SPDX-License-Identifier: LicenseRef-Super-Durable-1.0

import {
  gracefulComplete,
  jsonCodec,
  stringCodec,
  type Context,
  type Step,
  type StepDecision,
} from "../../src/index.js";

export interface ModelInput {
  value: number;
}

export const modelInputCodec = jsonCodec<ModelInput>({
  typeName: "ModelInput",
  decode: (value) => value as ModelInput,
});

export const dateCodec = jsonCodec<Date>({
  typeName: "Date",
  decode: (value) => new Date(String(value)),
  encode: (value) => value.toISOString(),
});

export const stringArrayCodec = jsonCodec<readonly string[]>({
  typeName: "string[]",
  decode: (value) => value as readonly string[],
});

export class CompleteStringStep implements Step<string> {
  public readonly inputCodec = stringCodec;

  public getStepType(): string {
    return "CompleteStringStep";
  }

  public execute(_context: Context, input: string): StepDecision {
    return gracefulComplete(input);
  }
}
